src/core/index.js

```diff
@@ -1,9 +1,38 @@
 import Vue from './instance/index'
 import { initGlobalAPI } from './global-api/index'
 import { isServerRendering } from 'core/util/env'
+import { warn } from 'core/util/index'
+import { modifierCode } from '../compiler/codegen/events'
```

Comment:
Modules under `/core` should not import from modules under `/compiler`. The only coupling between core and compiler happens in entry files.